src/client/mod.rs

```diff
@@ -18,43 +17,49 @@ use std::collections::BTreeMap;
 /// use std::env::var_os;
 /// use warpgrapher::client::graphql;
 ///
-/// let query = "query { Project { id name } }";
-/// let results = graphql("http://localhost:5000/graphql".to_owned(), query.to_owned(), None);
-/// let projects = results.unwrap().get("Project");
+/// #[tokio::main]
+/// async fn main() {
+///     let query = "query { Project { id name } }";
+///     let results = graphql("http://localhost:5000/graphql".to_owned(), query.to_owned(), None).await;
+///     let projects = results.unwrap().get("Project");
+/// }
 /// ```
-#[actix_rt::main]
+#[allow(clippy::needless_doctest_main)]
 pub async fn graphql(
     endpoint: String,
     query: String,
     input: Option<Value>,
 ) -> Result<Value, Error> {
```

Comment:
The documentation comment should describe the conditions under which an error is returned.